### PR TITLE
✨ Settings

### LINKED ISSUE #1092: Feedback from Nico P.
Image

- [ ] Cuando ingresas desde una Tablet, a veces la UI se rompe un poquito. Especialmente las columnas de Monto depositado, Valor, TNA y especialmente el porcentaje de tokens OP. Adjunto captura también (imagen 1)

Image

- [ ] Sería bueno que el texto que explica qué es el "Factor de salud" en la sección Panel de control se muestre a la derecha que tiene más lugar en vez de hacia abajo. Dejo una captura de ejemplo (imagen 2)

### PR DESCRIPTION
**Pull request by JuampiRombola**
closes #1092 

image

image

### CODE REVIEW

File: components/AdvancedSwitch/index.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React, { FC } from 'react';
+import { Box, Typography } from '@mui/material';
+import { useCustomTheme } from 'contexts/ThemeContext';
+import { useTranslation } from 'react-i18next';
+import Switch from 'components/Switch';
+
+type Props = {
+  sx?: React.ComponentProps<typeof Box>['sx'];
```
Comment (jgalat) [inline components/AdvancedSwitch/index.tsx:8]: You can import `BoxProps`

File: components/DropdownMenu/index.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ interface Props<T> extends Pick<MenuProps, 'anchorOrigin' | 'transformOrigin'> {
   renderOption: (value: T) => ReactNode;
   'data-testid'?: string;
   disabled?: boolean;
+  buttonSx?: React.ComponentProps<typeof Button>['sx'];
```
Comment (jgalat) [inline components/DropdownMenu/index.tsx:13]: I believe you can import `ButtonProps` from mui

File: components/Settings/index.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => setAnchorEl(event.currentTarget);
+  const handleClose = () => setAnchorEl(null);
+
+  return (
+    <>
+      <IconButton
+        id="view-contracts-button"
```
Comment (jgalat) [inline components/Settings/index.tsx:20]: this id is duplicated, consider using another one

File: components/Settings/index.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        aria-controls={open ? 'basic-menu' : undefined}
+        aria-haspopup="true"
+        aria-expanded={open ? 'true' : undefined}
+        onClick={handleClick}
+        size="small"
+        sx={{ outline: '1px solid', width: '32px', height: '32px', borderRadius: '50%', p: '0px' }}
+        // variant="outlined"
```
Comment (jgalat) [inline components/Settings/index.tsx:27]: remove comment

File: components/Settings/index.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      >
+        <SettingsRoundedIcon sx={{ fontSize: 20 }} />
+      </IconButton>
+      <Menu
+        id="basic-menu"
```
Comment (jgalat) [inline components/Settings/index.tsx:32]: this id is duplicated, consider using another one. Remember to also update the `aria-control` attr above with the same

**Review by sebipap — COMMENTED** [review]
lgtm, left minor comments

File: components/Settings/index.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+        MenuListProps={{
+          'aria-labelledby': 'view-contracts-button',
```
Comment (jgalat) [inline components/Settings/index.tsx:37]: Use same id as its label elemnt

**Review by jgalat — APPROVED** [review]
lgtm! minor comments

**Review by franm91 — DISMISSED** [review]

File: components/AdvancedSwitch/index.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import React, { FC } from 'react';
+import { Box, BoxProps, Typography } from '@mui/material';
+import { useCustomTheme } from 'contexts/ThemeContext';
+import { useTranslation } from 'react-i18next';
+import Switch from 'components/Switch';
+
+type Props = {
+  sx?: BoxProps;
```
[the remark this answers is not in the document]
Comment (sebipap) [reply components/AdvancedSwitch/index.tsx:8]: +1, either jorge's approach or `SxProps<Theme>`

**Review by sebipap — APPROVED** [review]

**Review by jgalat — APPROVED** [review]

**Review by franm91 — APPROVED** [review]

**Pull request merged**